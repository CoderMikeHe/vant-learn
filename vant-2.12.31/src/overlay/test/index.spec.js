import { mount } from '../../../test';
import Overlay from '..';

test('z-index prop', () => {
  const wrapper = mount(Overlay, {
    propsData: {
      show: true,
      zIndex: 99,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('class-name prop', () => {
  const wrapper = mount(Overlay, {
    propsData: {
      show: true,
      className: 'my-overlay',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('custom style prop', () => {
  const wrapper = mount(Overlay, {
    propsData: {
      show: true,
      customStyle: {
        backgroundColor: 'red',
      },
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('duration prop', () => {
  const wrapper = mount(Overlay, {
    propsData: {
      show: true,
      duration: 1,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('click event', () => {
  const onClick = jest.fn();
  const wrapper = mount(Overlay, {
    context: {
      on: {
        click: onClick,
      },
    },
  });

  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('default slot', () => {
  const wrapper = mount(Overlay, {
    scopedSlots: {
      default: () => 'Custom Default',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('lock-scroll prop', () => {
  const onTouchMove = jest.fn();
  const wrapper = mount({
    template: `
      <div @touchmove="onTouchMove">
        <van-overlay :lock-scroll="lockScroll" />
      </div>
    `,
    data() {
      return {
        lockScroll: true,
      };
    },
    methods: {
      onTouchMove,
    },
  });

  wrapper.find('.van-overlay').trigger('touchmove');
  expect(onTouchMove).toHaveBeenCalledTimes(0);

  wrapper.setData({ lockScroll: false });
  wrapper.find('.van-overlay').trigger('touchmove');
  expect(onTouchMove).toHaveBeenCalledTimes(1);
});
