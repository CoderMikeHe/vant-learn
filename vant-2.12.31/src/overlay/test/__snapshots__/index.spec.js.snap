// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class-name prop 1`] = `<div class="van-overlay my-overlay" name="van-fade"></div>`;

exports[`custom style prop 1`] = `<div class="van-overlay" style="background-color: red;" name="van-fade"></div>`;

exports[`default slot 1`] = `<div class="van-overlay" style="display: none;" name="van-fade">Custom Default</div>`;

exports[`duration prop 1`] = `<div class="van-overlay" style="animation-duration: 1s;" name="van-fade"></div>`;

exports[`z-index prop 1`] = `<div class="van-overlay" style="z-index: 99;" name="van-fade"></div>`;
